### DISCUSSION TITLE
errors with IRAF installation

### DISCUSSION THREAD

**Discussion by kevinluhman** [iraf-community/iraf#312] [Q&A]
I am trying to install IRAF in macOS Ventura 13.4.1 and am encountering errors.
Here are the commands that I ran:

tar zxf iraf-2.17.tar.gz
./install
make macintel
make sysgen 2>&1 | tee build.log

Based on an error with the last step and a remedy suggested in another query about
that error, I added "add #include <unistd.h>" at the top of unix/f2c/src/sysdep.c
That step then proceeds farther, but it still produces errors and IRAF isn't installed.
Here is the log:

[build.log](https://github.com/iraf-community/iraf/files/12003119/build.log)

Can anyone suggest a solution? Thanks!

**Comment by olebole** [comment]
The problem is that with the newest XCode, the compilations of 2.17 does not work. Until a new release is out, you can use the latest development code by downloading https://github.com/iraf-community/iraf/archive/refs/heads/main.tar.gz. Note that the installation procedure changed; see [here](https://github.com/iraf-community/iraf/blob/main/INSTALL.md) (or the `INSTALL.md` file in the tarball).

**Comment by kevinluhman** [comment]
Thanks for your quick response. When I run the make command, I receive the following errors. Can you suggest a solution?

[build.log](https://github.com/iraf-community/iraf/files/12016434/build.log)

**Reply by olebole** [reply]
Did you start from a freshly unpacked tarball? And can you give the result of
```shell
unix/hlib/irafarch.sh -current
ls -l bin unix/bin noao/bin
```

**Reply by kevinluhman** [reply]
Yep, I used the fresh unpacked tarball. Here are the results from those commands:
->
lrwxr-xr-x  1 kll207  PSU\Domain Users  12 Jul 11 11:33 bin -> bin.macintel
lrwxr-xr-x  1 kll207  PSU\Domain Users  12 Jul 11 11:33 noao/bin -> bin.macintel
lrwxr-xr-x  1 kll207  PSU\Domain Users  12 Jul 11 11:33 unix/bin -> bin.macintel

**Reply by olebole** [reply]
It looks like that the `irafarch.sh` script doesn't like the space in your group name. You can try with 
```shell
export IRAFARCH=macintel
make
```

**Reply by kevinluhman** [reply]
Thanks, "make" seemed to work okay now, and "make test" was okay as well.
Should those steps automatically produce a .iraf directory? Do I need to run mkiraf somewhere?
Is there anything else to do before IRAF can be run? Thanks again for your help!

**Reply by olebole** [reply]
You should either do a system installation with `sudo make install`, or do an "in-place installation" with `make inplace` (make sure that `IRAFARCH` is set in both cases) as described in the [INSTALL.md](https://github.com/iraf-community/iraf/blob/main/INSTALL.md#install-the-software-at-its-final-place) file.

**Reply by kevinluhman** [reply]
Sorry, I had forgot about "make inplace". I have now run that. If I run "ecl" in an xterm window within iraf-main, an IRAF terminal appears. Does that need to be running in an xgterm window? If so, how do I create an xgterm window? The command "xgterm" doesn't seem to be recognized. Also, if I run an IRAF task in the terminal, it says that the task has no param file. Sorry for all of the questions!

**Reply by olebole** [reply]
xgterm is not part of IRAF, but part of x11iraf, which must be installed separately. You can also use xterm, but the graphics are not so nice. If you have an old `login.cl`, you need to create the local uparm dir with `mkiraf`.

**Reply by kevinluhman** [reply]
When I run mkiraf in a new iraf directory, I get the following errors. Is this related to the fact that I did an "inplace" installation?

cat: /etc/iraf/irafroot: No such file or directory
Error: current directory is not an iraf user login directory

**Reply by olebole** [reply]
Oh, yes, this is a bug in the **mkiraf** script; I just created Pull Request #314 to fix this.
In the meantime you could set the **iraf** environment variable by hand (adjust as needed; make sure the trailing slash exists):
```shell
export iraf=/Users/kll207/iraf-2.17/
```

**Reply by kevinluhman** [reply]
I used "export iraf=/Users/kll207/iraf-main/" and ran mkiraf. Here's the output:

-- creating a new uparm directory
A new LOGIN.CL file has been created in the current directory.
You may wish to review and edit this file to change the defaults.

However, it didn't create a login.cl file. It only created an empty uparm directory.

**Reply by olebole** [reply]
This is just a remnant of the old script. Do you still the missing parameter issue?

The login.cl is only created if you use the `-c` option; otherwise the default one is used.

**Reply by kevinluhman** [reply]
Running mkiraf -c produces the login.cl, and iraf now seems to work, thanks!
I have now installed x11iraf-2.1 so that I can use xgterm. When I run "tic xgterm/xgterm.terminfo" it doesn't seem like anything happens.
The value of TERM is still "xterm" when spawning an xgterm window. Any suggestions?

**Reply by kevinluhman** [reply]
I tried creating .terminfo directory in my home directory and running "tic xgterm/xgterm.terminfo" but now it says "Segmentation fault: 11"

**Reply by olebole** [reply]
A crash is most probably a problem with your "tic"; I just tried it on both a (latest) Mac and a Linux PC, without crash (both created files in `~/.terminfo`).
You can just set the TERM manually before starting IRAF.

**Reply by kevinluhman** [reply]
I was getting the error with tic in bash, but for some reason tic now works properly when running it in tcsh. xgterm now works. Thanks for your help!

**Comment by judychang2023** [comment]
The error occurs when I compile the iraf-2.17 to my macOS system x86_64 (Catalina 10.15.7).
Te error shows as below:
ld: file too small (length=8) file '/TO_THE_PATH/iraf-2.17/lib/libsys.a' for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [sysgen] Error 1

I removed the file libsys.a, make clean, and make macintel. 
And tried again " make sysgen 2>&1 | tee build.log", the same error still appears.
How can I solve the problem?
Thank you for the help~

**Reply by olebole** [reply]
I would recommend to use version 2.17.1, which has a greatly simplified installation procedure. If it still does not work, please do not forget to attach the build log.
